src/spi.rs

```diff
@@ -408,16 +359,19 @@ impl spi::SpiDevice for Mock {
         for op in operations {
             match op {
                 Operation::Read(buffer) => {
-                    SpiBusRead::read(self, buffer)?;
+                    SpiBus::read(self, buffer)?;
                 }
                 Operation::Write(buffer) => {
-                    SpiBusWrite::write(self, buffer)?;
+                    SpiBus::write(self, buffer)?;
                 }
                 Operation::Transfer(read, write) => {
-                    spi::SpiBus::transfer(self, read, write)?;
+                    SpiBus::transfer(self, read, write)?;
                 }
                 Operation::TransferInPlace(buffer) => {
-                    spi::SpiBus::transfer_in_place(self, buffer)?;
+                    SpiBus::transfer_in_place(self, buffer)?;
+                }
+                Operation::DelayUs(_) => {
+                    todo!();
```

Comment:
@Dirbaio: you introduced this new operation in https://github.com/rust-embedded/embedded-hal/commit/d07d39e359ed2157cbc0bbd88ef950b60c58e804
could you please advise on how this is best covered here? (sorry, i'm not at all familiar with the SPI implementation - i just want to bump this to the newest e-h version before the release of e-h-m for 1.0.0-alpha)

thanks!

Reply:
In non-mock code you'd either make it generic on a `T: DelayUs`, or if the implementation is hardware-specific directly use a hardware-specific delay. If for whatever reason you can't, returning a "delay not supported" error is also an option.

In the mock case you might want to check it against the expectations? so the user can assert the delay is done. Not sure.

Reply:
For the case of making it generic over `T: DelayUs`, this crate provides a no-op delay mock, which one could use for this.